### PR TITLE
dev-lua/mpack: Bump to 1.0.3

### PR DESCRIPTION
**Pull request by lluixhi**
Switch to the Makefile because upstream added the ability to select the system lua implementation instead of downloading and compiling lua, luarocks, busted, etc.

### CODE REVIEW

**Comment by kensington** [comment]
@gentoo/proxy-maint

**Comment by kensington** [comment]
This looks pretty good, however there's one small issuecaused by switching to the Makefile:

``cc -shared -march=native -O2 -pipe -fPIC -DMPACK_DEBUG_REGISTRY_LEAK  -Wl,-O1 -Wl,--as-needed -Wl,--hash-style=gnu lmpack.c -o mpack.so -llua -lm``

We're not allowed to call ``cc`` directly, but it's pretty easy to fix - either add an extra argument in the build phase (``CC="$(tc-getCC)"``) or export ``CC`` prior to building (``tc-export CC``).

File: dev-lua/mpack/mpack-1.0.3.ebuild [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+# $Id$
+
+EAPI=6
+
+inherit toolchain-funcs
+
+DESCRIPTION="Lua bindings for libmpack"
+HOMEPAGE="https://github.com/tarruda/libmpack/"
+SRC_URI="https://github.com/tarruda/libmpack/archive/${PV}.tar.gz -> ${P}.tar.gz"
+S="${WORKDIR}/libmpack-${PV}/binding/lua"
```
Comment (kensington) [inline dev-lua/mpack/mpack-1.0.3.ebuild]: By convention, ``S`` normally appears later in the ebuild:

```
DEPEND="app-foo/bar"
RDEPEND="${DEPEND}"

S="${WORKDIR}/baz"

src_compile() {
```

That said, there there is no formal policy about the placement of ``S`` (and this is no regression compared to 1.0.2) so this comment is purely informational.

**Comment by lluixhi** [comment]
Alright, I added `CC=$(tc-getCC)` to the arguments passed to emake during the compile phase.

I didn't realize that was an issue, thanks.

**Comment by kensington** [comment]
Look good, thanks! Picked as 6d02ddc7257da822d21232197926256c683842dc.

**Pull request closed without merging**